File: ai/teenager.py
# Class worked on by: Claire
import random
import string
from personality import Personality
from data_pre_processor import DataPreProcessor


class Teenager(Personality):

    def __init__(self, name):
        Personality.__init__(self, name)
        self.age = random.randint(13, 21)
        self.probSpellingMistake = random.randint(1,5)
        

    def addPersonality(self, input):
        
        output = self.shortenWords(input)
        return output

    # You -> u for instance, more likely to be commonly appropriate. Needs to return a string. - Kishore
    def shortenWords(self, input):
        shortWords = {
	        'your' : 'ur',
            'tonight' : '2nite',
            'for your information' : 'fyi',
            'okay' : 'k',
            'ok':'k',
	        'to' : '2',
	        'tomorrow' : '2moro',
            'easy' : 'ez',
            'see you' : 'cya',
            'because' : 'cuz',
            'you' : 'u',
            }       
        preProcessor = DataPreProcessor(input) 
        input = preProcessor.string2Array(input)    
        for x in range(0,len(input)):
            for y in shortWords:
                if(str(input[x]) == y):
                    input[x] = shortWords[y]
        input = preProcessor.array2String(input)               
        
        return input

File: client/TheTuringGameDemo/app/src/main/java/com/sweng/theturinggamedemo/Teenager.java
// Class worked on by the AI Group: Claire McNamara

package com.sweng.theturinggamedemo;

import java.util.HashMap;
import java.util.Random;

public class Teenager extends BotPersonality {

    public int age, probSpellingMistake;

    public Teenager(String name){
        super(name);
        Random rand = new Random();
        this.age = rand.nextInt((21 - 13) + 1) + 13;
        this.probSpellingMistake = rand.nextInt((5 - 1) + 1) + 5;
    }

    @Override
    public String addPersonality(String input){
        input = shortenWords(input);
        return input;
    }

    // Adds in slang into the AI's message.
    private String shortenWords(String input){
        HashMap<String, String>  shortenedWords = new HashMap<>();
        shortenedWords.put("your", "ur");
        shortenedWords.put("tonight", "2nite");
        shortenedWords.put("for your information", "fyi");
        shortenedWords.put("okay", "k");
        shortenedWords.put("ok", "k");
        shortenedWords.put("to", "2");
        shortenedWords.put("tomorrow", "2moro");
        shortenedWords.put("easy", "ez");
        shortenedWords.put("because", "cuz");
        shortenedWords.put("you", "u");

        String[] inputAr = input.split(" ");
        int wordIndex = 0;
        for (String word:inputAr) {
            if (shortenedWords.containsKey(word)){
                inputAr[wordIndex] = shortenedWords.get(word);
            }
            wordIndex++;
        }
        input = String.join(" ", inputAr);
        return input;
    }
}